x-pack/elastic-agent/pkg/agent/install/perms_windows.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build windows
+
+package install
+
+import (
+	"io/fs"
+	"path/filepath"
+
+	"github.com/hectane/go-acl"
+	"golang.org/x/sys/windows"
+
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/application/paths"
+)
+
+// fixPermissions fixes the permissions so only SYSTEM and Administrators have access to the files in the install path
+func fixPermissions() error {
+	return recursiveSystemAdminPermissions(paths.InstallPath)
+}
+
+func recursiveSystemAdminPermissions(path string) error {
+	return filepath.Walk(path, func(name string, info fs.FileInfo, err error) error {
+		if err == nil {
+			// first level doesn't inherit
+			inherit := true
+			if path == name {
+				inherit = false
+			}
+			err = systemAdministratorsOnly(name, inherit)
+		}
+		return err
+	})
+}
+
+func systemAdministratorsOnly(path string, inherit bool) error {
+	systemSID, err := windows.StringToSid("S-1-5-18")
+	if err != nil {
+		return err
+	}
+	administratorsSID, err := windows.StringToSid("S-1-5-32-544")
+	if err != nil {
+		return err
+	}
+	return acl.Apply(
+		path, true, inherit,
+		acl.GrantSid(0xF10F0000, systemSID), // full control of all acl's
```

Comment:
What is `0xF10F0000`?

Reply:
That is the hex needed for the "full control of all acl's".

Reply:
Can you left a comment describing Windows cryptic ways of specifying things